grpclb/src/main/java/io/grpc/grpclb/GrpclbLoadBalancerProvider.java

```diff
@@ -77,33 +79,38 @@ public ConfigOrError parseLoadBalancingPolicyConfig(
   ConfigOrError parseLoadBalancingConfigPolicyInternal(
       Map<String, ?> rawLoadBalancingPolicyConfig) {
     if (rawLoadBalancingPolicyConfig == null) {
-      return ConfigOrError.fromConfig(DEFAULT_MODE);
+      return ConfigOrError.fromConfig(GrpclbConfig.create(DEFAULT_MODE));
+    }
+    Object rawTarget =
+        rawLoadBalancingPolicyConfig.get(GrpclbLoadBalancerProvider.SERVICE_CONFIG_TARGET_NAME);
+    String target = null;
+    if (rawTarget instanceof String) {
```

Comment:
If `rawTarget` is not a string, an error should be returned instead of silently ignored. You could either cast without checking, or add an else-branch to return the error.

Reply:
`GrpclbLoadBalancer#retrieveGrpcLbConfig` always expects valid config even if it is invalid. this was trying matching the same behavior. just like childPolicy mode falls back to DEFAULT_MODE if unknown mode is passed which should be error? I can see we at least should have same type, but it is still weird.

Reply:
The current behavior you described was implemented before Service Config Error Handling was designed. We may want to revisit it now. @ejona86 may have some opinions.

Reply:
Failing if wrong type sounds correct, but I'm not familiar with the two different code paths here (why there are two code paths; I can tell one is old and one is new, but I'm still surprised). So I don't know if this change change short-term or long-term.

Reply:
both are old code ways, the only difference i can tell is one logs and the other doesn't.
in long term, we should throw and handle service config error instead of falling back to default. given grpclb's status i am not sure it will happen or not.

Reply:
No, this file is the new code path, and the one in `GrpclbLoadBalancer` is the old one. The new one is not used until the new `ManagedChannelImpl2` is used (#6564). Once the old `ManagedChannelImpl` is replaced by the new one, the old path can be deleted. I expected this to happen soon as we just need to gain confidence from internal testing and integration testing.

Reply:
i migrated the parsing in LB, and made this parse **very strict**. following suggestions involves behavior changes, and i want to get some feedback. @ejona86 

- return `defaultChildPolicy` (round robin?) if no child policy specified. we still should return Error if the `childPolicy` contains only unknown policies.
- previously, it doesn't ignore unknown policy. it falls back to default but it didn't fall back to next specified policy in the `childPolicy`. Since service config error handling design skips unknown lb policy, it should follow the same logic. It should skip unknown until it sees supported child policies (pick first and round robin)

Reply:
i'll summarize this change in other comment.